app/src/main/java/uk/gov/android/authentication/login/AppAuthSession.kt

```diff
@@ -3,11 +3,13 @@ package uk.gov.android.authentication.login
 import android.content.Context
 import android.content.Intent
 import androidx.activity.result.ActivityResultLauncher
+import net.openid.appauth.AuthorizationException
 import net.openid.appauth.AuthorizationResponse
 import net.openid.appauth.AuthorizationService
 
 class AppAuthSession(
     context: Context,
+    private val clientAuthenticationProvider: ClientAuthenticationProvider = ClientAuthenticationProviderImpl()
```

Comment:
I think we can move this from the constructor now considering we're not mocking it in any of the tests?